packages/web3-eth-ens/src/contracts/Registry.js

```diff
@@ -39,10 +38,7 @@ function Registry(ens) {
     var self = this;
     this.ens = ens;
     this.contract = ens.checkNetwork().then(function (address) {
-        var contract = new Contract(REGISTRY_ABI, address);
-        contract.setProvider(self.ens.eth.currentProvider);
-
-        return contract;
+        return new self.ens.eth.Contract(REGISTRY_ABI, address);
```

Comment:
I was able to remove the additional ``setProvider`` call because of the fixes I've applied to the ``RequestManager``.

Reply:
Ok! Fair enough.